### DISCUSSION TITLE
Issue with Pagination in Artifact Hub API

### DISCUSSION THREAD

**Discussion by Ashken12** [artifacthub/hub#4110] [General]
Hello,

I am experiencing an issue with the Artifact Hub API while trying to retrieve package search results. Specifically, when I attempt to fetch multiple pages of results using the pagination parameters (page), I consistently receive the same set of results regardless of the page number specified.

Here are the details of my API requests:

Base URL: https://artifacthub.io/api/v1/packages/search
Query Parameter: ts_query_web=mysql
Limit: 60 results per page
For example, both of the following requests return the same results:

Page 1: https://artifacthub.io/api/v1/packages/search?limit=60&page=1&ts_query_web=mysql
Page 2: https://artifacthub.io/api/v1/packages/search?limit=60&page=2&ts_query_web=mysql
I’ve checked the API documentation, but I couldn’t find any information indicating that the pagination is restricted in this manner.

Could you please provide insight into whether this is a known issue, or if there are any specific guidelines for properly implementing pagination in API requests?

Thank you for your help!

**Comment by tegioz** [comment]
Hi @Ashken12 👋

The `page` parameter is not supported by that endpoint, you need to use `offset` and `limit`. For more details please take a look at the [endpoint documentation](https://artifacthub.io/docs/api/#/Packages/searchPackages).

**Reply by Ashken12** [reply]
Thankyou @tegioz
